contrib/ibmock/lib.h

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef __LIB_H
+#define __LIB_H
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <pthread.h>
+#include <sys/uio.h>
+
+
+struct array {
+    void   *data;
+    size_t elem_size;
+    int    count;
+};
+
+pthread_mutex_t global_lock = PTHREAD_MUTEX_INITIALIZER;
+
+static inline void lock(void)
```

Comment:
IMO it's better to not have these functions lock and unlock, better call pthread locks on global variable - it's easier to follow IMO